File: mobile/android/oxo-android-ben20/src/app/src/main/java/com/ostorlab/zip/viewer/MainActivity.java
package com.ostorlab.zip.viewer;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button uploadBtn = findViewById(R.id.btn_upload);
        Button viewBtn = findViewById(R.id.btn_view);

        uploadBtn.setOnClickListener(v -> startActivity(new Intent(this, ZipUploadActivity.class)));
        viewBtn.setOnClickListener(v -> startActivity(new Intent(this, ZipViewerActivity.class)));
    }
}

File: mobile/android/oxo-android-ben75/src/app/src/main/java/com/documentmanager/MainActivity.kt
package com.documentmanager

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.documentmanager.data.DocumentDataManager

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        
        // Initialize document data on app startup
        DocumentDataManager.initializeDocuments(this)
        
        setupUI()
    }
    
    private fun setupUI() {
        findViewById<Button>(R.id.btn_my_documents).setOnClickListener {
            startActivity(Intent(this, DocumentsActivity::class.java))
        }
        
        findViewById<Button>(R.id.btn_share_documents).setOnClickListener {
            startActivity(Intent(this, ShareActivity::class.java))
        }
        
        findViewById<Button>(R.id.btn_settings).setOnClickListener {
            startActivity(Intent(this, SettingsActivity::class.java))
        }
    }
}
